aws-ts-ec2-provisioners/provisioners/index.ts

```diff
@@ -15,7 +15,12 @@ export interface ConnectionArgs {
     username?: pulumi.Input<string>;
     password?: pulumi.Input<string>;
     privateKey?: pulumi.Input<string>;
-    privateKeyPassphrase?: pulumi.Input<string>;
+	privateKeyPassphrase?: pulumi.Input<string>;
+	/**
+     * A string value to control the replacement of the provisioner during each update.
+     * Provide a stable value if you do not need the provisioner to run each time you run an update.
+     */
+    changeToken: string;
```

Comment:
I feel like this doesn't quite belong on the `ConnectionArgs` - and might make more sense as a property on `CopyFile` args and `RemoteExec` args?  In terraform, this is often called `keepers` - perhaps use that terminology (I've never found any terminology I particularly like for this).

Note that for the specific case of changing when the file contents change - it might be nice to actually have a first class `hash` property.  Doing just that might be sufficient here?

Though looking at your original example - perhaps the reason you did it the way you did was to trigger both of these to re-run - which would not be possible with my suggestion above due to the lack of https://github.com/pulumi/pulumi/issues/838.

Feel free to keep this as is - but perhaps consider `keepers` as the name?

As for doc comment here - the inconsistency is a little jarring - perhaps add comments on all properties? And either adopt the same `//` style, or update all existing doc comments to `/** */` style?

Reply:
> I feel like this doesn't quite belong on the ConnectionArgs - and might make more sense as a property on CopyFile args and RemoteExec args?

Yeah I agree -- it was a little hacky to put it with the `ConnectionArgs`, to be honest. I moved it to be a property of those component resources, as well as the provisioner dynamic resource's "state" input as well.

> Though looking at your original example - perhaps the reason you did it the way you did was to trigger both of these to re-run

Yes, that's correct. The user can control which one they would like to re-run every time the file(s) content changes.

> Feel free to keep this as is - but perhaps consider keepers as the name?

I don't like the name `keepers`, so I left the name `changeToken`, but can change it if you insist on using `keepers`.